saq/queue/postgres.py

```diff
@@ -484,7 +484,29 @@ async def iter_jobs(
 
     async def abort(self, job: Job, error: str, ttl: float = 5) -> None:
         job.error = error
-        await self.update(job, status=Status.ABORTING)
+
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                SQL(
+                    dedent(
+                        """
+                        SELECT status
+                        FROM {jobs_table}
+                        WHERE key = %(key)s
+                        FOR UPDATE
```

Comment:
do we use for update everywhere else? is this needed given the whole
block is in a transaction?

Reply:
why don’t we have for update in sweep?

Reply:
I used FOR UPDATE here because if a job that hasn't been dequeued yet is aborted, I didn't want dequeue to accidentally pick it up in between the select and the update. As for sweep, do you think we should? I figured if a job's stuck or orphaned, there's no need to lock it. And if it's not, we don't want to lock them in case a worker needs to update it.